src/map/prepareLayer.js

```diff
@@ -244,6 +252,21 @@ function readData(mapId, layer, dispatch, doUpdateTsLayer) {
       renderData(mapId, layerObj, dispatch, doUpdateTsLayer);
     });
   }
+  if (fileType === 'superset') {
+    const currentState = dispatch(getCurrentState());
+    const config = {
+      endpoint: 'slice',
+      extraPath: sourceURL['slice-id'],
+      base: currentState.APP && currentState.APP.supersetBase,
+    };
+
+    superset.api.fetch(config, // fetch with config
+      (res) => res) // pass in callback func to process response
+      .then(data => {
+        layerObj.source.data = superset.processData(data); // assign processed data to layerObj
```

Comment:
i think here, in case we add filters to the superset layer we should generate the filter options before passing the `layerObj` into `renderData` function i.e 
```
if (layerObj.aggregate && layerObj.aggregate.filter) {
    layerObj.filterOptions = generateFilterOptions(layerObj);
}
```

Reply:
oooh good idea, i'll add this

Reply:
yeah good call, added the aggregation function too